ir/src/lib.rs

```diff
@@ -10,9 +10,9 @@ use symbol_table::{IdentifierType, SymbolTable};
 pub mod boundary_constraints;
 use boundary_constraints::BoundaryConstraints;
 
-pub mod transition_constraints;
-use transition_constraints::{AlgebraicGraph, TransitionConstraints, MIN_CYCLE_LENGTH};
-pub use transition_constraints::{NodeIndex, TransitionConstraintDegree};
+pub mod integrity_constraints;
+use integrity_constraints::{AlgebraicGraph, IntegrityConstraints, MIN_CYCLE_LENGTH};
+pub use integrity_constraints::{NodeIndex, TransitionConstraintDegree};
```

Comment:
I think we should probably rename `TransitionConstraintDegree` as well, to keep everything in the IR consistent. That would mean updating all of the related comments, etc. as well. It seems like a deliberate decision that you chose not to do this though, so it would be good to understand your reasoning & discuss

Reply:
I wasn't sure if we wanted to change this naming since this struct is defined in winterfell so I wasn't sure if we should keep it the same in IR as well or not. But I think it makes sense to change this in IR. Thanks.

Reply:
Cool, I figured that was your thinking (and it makes sense). Sounds good to change it for our IR though